### DISCUSSION TITLE
Options picker showing list of values from a text field

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9306] [Help] [answered]
How can I populate an options picker with the list of value in a text field? Now I only get the value of the current record ...
![Screenshot 2023-01-11 at 09 35 01](https://user-images.githubusercontent.com/4005405/211758025-9491d740-56bc-4b24-bff7-e0ad86fe8bfe.png)

**Comment by ecxs** [comment]
Can anybody please confirm whether I'm doing something wrong, or whether this is actually a bug? These are my settings for the Options Picker:
![Screenshot 2023-01-12 at 13 49 47](https://user-images.githubusercontent.com/4005405/212070758-11e15772-4344-4890-9235-613d4878c85f.png)

**Comment by melohagan** [comment]
Hey @ecxs 

You can see that the source of the Options Picker is **Dataprovider**, on column *User*.

Make sure that the Data provider returns the data that you expect. You could place a paragraph under the data provider with the binding `{{ Dataprovider.rows }}` to check.

**Reply by ecxs** [reply]
Hi @melohagan thanks for taking the time to help me out, as well as for the tip! I used the 'Paragraph /  {{ Dataprovider.Rows }}' trick, and now realise that the Dataprovider is my side table is limited to 1 record. And the Dataprovider in my Table is limited as well (by Pagination).

So let me rephrase my question: how can I get a dynamic drop down list on all the (unique) values in the 'User' field in my 'Assets' table?

P.s.: the {{ Dataprovider.Rows }} is case sensitive and therefore {{ Dataprovider.rows }} (with a small r) did not work at first ;-)

![Screenshot 2023-01-23 at 13 37 32](https://user-images.githubusercontent.com/4005405/214042720-fba073bf-7068-4af1-9d8b-14ab6cfa2d46.png)

**Comment by ecxs** [comment, marked as answer]
Hi @melohagan I think I'm one step closer to a solution ... I've created a Query 'ValueList_Users.Rows' with the following query:

`SELECT DISTINCT User FROM Assets`

And added this a a datasource to my Details side panel., and it's visible via 'Paragraph / {{ ValueList_Users.Rows }}'.

However, I can not select this datasource as the Options Provider. Am I doing something wrong, or is this not possible (yet)?

Thanks again!

**Reply by melohagan** [reply]
That's a good start! The next step would be to include a Data provider that uses that query above your field group. Something like so:

> Form
> > Container
> > > Data Provider (unique users query datasource, give it a name like: *Users provider*)
> > > > Fieldgroup

Now you'll be able to access the *Users provider* as the data provider source of your options picker

**Reply by ecxs** [reply]
@melohagan Thanks for your explanation! I now got it to work :-) ... Is there any documentation that explain where Data Providers need to be located?

**Reply by melohagan** [reply]
There is a bit of information on the Data provider here: https://docs.budibase.com/docs/data-provider

Basically the Data Provider can be located anywhere, and anything underneath will be able to access that data.
Just be careful not to include Data providers within repeaters if possible as this can definitely cause performance issues.

**Reply by ecxs** [reply]
@melohagan Thanks again. I noticed that underneath means inside ...

This doesn't work:

![Screenshot 2023-01-24 at 12 38 32](https://user-images.githubusercontent.com/4005405/214282382-b81b5088-491c-4648-9427-84220a9a5111.png)

And this does:

![Screenshot 2023-01-24 at 12 39 03](https://user-images.githubusercontent.com/4005405/214282410-5de0d249-b982-415d-ab60-8188732ed284.png)

I also noticed that I had to increase the Limit on the Dataprovider in order to get all values in. Is there a method to set this to infinite, or should I just set it to a large enough number?

**Reply by melohagan** [reply]
I believe 5000 is the maximum limit. If you remove the limit, it may automatically take the max limit.

**Reply by ecxs** [reply]
When you remove the limit, it automatically switched back to 0. And when I enter 1000000 it remains 1000000 ...
